From 65341c4143671740841b02603dd20c2e513525ba Mon Sep 17 00:00:00 2001
From: Roy Pledge <Roy.Pledge@freescale.com>
Date: Wed, 2 Sep 2015 17:00:54 -0400
Subject: [PATCH 090/146] fsl_qbman: Fix Frame Queue Stash setup

The stashing subfields of the FQD context_a need to be populated so
that they are in proper byteorder once the entire context is swapped

Signed-off-by: Roy Pledge <Roy.Pledge@freescale.com>
---
 include/linux/fsl_qman.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/fsl_qman.h b/include/linux/fsl_qman.h
index 6dd7b48..5a68042 100644
--- a/include/linux/fsl_qman.h
+++ b/include/linux/fsl_qman.h
@@ -528,8 +528,8 @@ struct qm_mr_entry {
  * representation. */
 struct qm_fqd_stashing {
 	/* See QM_STASHING_EXCL_<...> */
-	u8 exclusive;
 #if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+	u8 exclusive;
 	u8 __reserved1:2;
 	/* Numbers of cachelines */
 	u8 annotation_cl:2;
@@ -540,6 +540,7 @@ struct qm_fqd_stashing {
 	u8 data_cl:2;
 	u8 annotation_cl:2;
 	u8 __reserved1:2;
+	u8 exclusive;
 #endif
 } __packed;
 struct qm_fqd_taildrop {
-- 
2.1.0.27.g96db324

